### PR TITLE
Mention minimum etcd versions

### PR DESCRIPTION
**Pull request by erictune**
Source: https://groups.google.com/d/msg/kubernetes-dev/jMPA4JzKiY4/HIx2ugvLBAAJ

### CODE REVIEW

File: content/en/docs/tasks/administer-cluster/configure-upgrade-etcd.md [hunk 1]
```diff
@@ -34,6 +34,12 @@ content_template: templates/task
 
 * Keeping stable etcd clusters is critical to the stability of Kubernetes clusters. Therefore, run etcd clusters on dedicated machines or isolated environments for [guaranteed resource requirements](https://github.com/coreos/etcd/blob/master/Documentation/op-guide/hardware.md#hardware-recommendations).
 
+* The minimum recommended versions of etcd to run in production are:
+  * `3.1.11+`
+  * `3.2.10+`
+  * `3.3.0+`
```
Comment (wojtek-t) [inline content/en/docs/tasks/administer-cluster/configure-upgrade-etcd.md]: Should we also mention k8s releases in which those versions are recommended?

E.g. 3.2.10+ from k8s releases 1.10+ (IIRC)

@jpbetz - thoughts?

Comment (jpbetz) [reply content/en/docs/tasks/administer-cluster/configure-upgrade-etcd.md]: Yes, maybe something like:

```
The minimum recommended versions of etcd to run in production are:
  * etcd `3.1.11+` for Kubernetes 1.8-1.9
  * etcd `3.2.10+` for Kubernetes 1.10+
```

?

**Comment by jpbetz** [comment]
Either that or something like this would be fine:

```
The minimum recommended versions of etcd to run in production are:
  * `3.1.11+`
  * `3.2.10+` (preferred)
```

**Comment by wojtek-t** [comment]
+1 for the above proposal by @jpbetz

**Comment by MikeSpreitzer** [comment]
Thanks for documenting this.

**Comment by erictune** [comment]
@jpbetz PTAL, not sure I captured what you asked for exactly right.  This is documenting for 1.11.

**Comment by jpbetz** [comment]
For 1.11 our main test pairing was k8s 1.11/etcd 3.2, so I'd like to shy away from any claims of 3.3 support. We officially claim backward compatibility for all 3.x etcd, and even 2.x (with a deprecation deadline coming up this year).  Maybe we should just include the 3.2 version here since this is about what we're recommending (as opposed to supporting) and 3.2 is the only version we have a strong direct test signal for for 1.11.

File: content/en/docs/tasks/administer-cluster/configure-upgrade-etcd.md [hunk 2]
```diff
@@ -388,6 +392,8 @@ TARGET_VERSION=2.2.1
 STORAGE_MEDIA_TYPE=application/json
 ```
 
+** Note: this upgrade procedure is for getting from 2.x to 3.x.  Version `3.0.17` is not recommended for running in production (see [prerequisites](#prereqisites) for minimum recommended etcd versions.
```
Comment (jpbetz) [inline content/en/docs/tasks/administer-cluster/configure-upgrade-etcd.md]: Looks great!

Comment (MikeSpreitzer) [reply content/en/docs/tasks/administer-cluster/configure-upgrade-etcd.md]: Why single out 3.0.17?  The guidance I had been seeing previously said just get it above 3.0.10.  https://groups.google.com/forum/#!topic/etcd-dev/nZQl17RjxHQ dumps on all of 3.0 --- and more emphatically than I am seeing here.  Somewhere there should be maintained an evergreen doc for general etcd use like https://groups.google.com/forum/#!topic/etcd-dev/nZQl17RjxHQ

Comment (jpbetz) [reply content/en/docs/tasks/administer-cluster/configure-upgrade-etcd.md]: Just saying `3.0` here sounds fine.

@MikeSpreitzer For the evergreen doc, that would be for the etcd docs? We should probably move discussion to github.com/etcd-io/etcd.

Comment (MikeSpreitzer) [reply content/en/docs/tasks/administer-cluster/configure-upgrade-etcd.md]: Yes, @jpbetz , I think the etcd docs should include a maintained statement like what you sent out in email.

Comment (erictune) [reply content/en/docs/tasks/administer-cluster/configure-upgrade-etcd.md]: I don't want to change the tested upgrade procedure.
I updated the advice to not  single out 3.0.17, and just suggest further upgrading to the recommended version of etcd.

**Comment by wojtek-t** [comment]
lgtm

File: content/en/docs/setup/scratch.md [hunk 3]
```diff
@@ -194,18 +194,13 @@ You have several choices for Kubernetes images:
...
-- Use images hosted on [Docker Hub](https://hub.docker.com/search/?q=etcd) or [Quay.io](https://quay.io/repository/coreos/etcd), such as `quay.io/coreos/etcd:v2.2.1`
-- Use etcd binary included in your OS distro.
-- Build your own image
-  - You can do: `cd kubernetes/cluster/images/etcd; make`
-
 We recommend that you use the etcd version which is provided in the Kubernetes binary distribution.   The Kubernetes binaries in the release
 were tested extensively with this version of etcd and not with any other version.
 The recommended version number can also be found as the value of `TAG` in `kubernetes/cluster/images/etcd/Makefile`.
 
+For the miniumum recommended version of etcd, please refer to
```
Comment (zacharysarah) [inline content/en/docs/setup/scratch.md]: @erictune 👋 Thanks for making these updates! For future reference, omit "please" per the [style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language). 

/cc @tengqm

**Pull request merged**